### DISCUSSION TITLE
Yolov5 predicting single class

### DISCUSSION THREAD

**Discussion by Una865** [ultralytics/yolov5#7902] [General]
I am training the YOLOv5 model on predicting Sign Language. However, the I train the model it outputs many classes for one single image(one sign shown) and I want to have prediction based on the most confident class and have only that class shown. How can I do that?

**Comment by glenn-jocher** [comment]
@Una865 you can modify max detections parameter down to 1

**Reply by Una865** [reply]
How can I do that in Google Colab, I cloned the GitHub yolov5

**Reply by glenn-jocher** [reply]
See https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb?hl=en

**Reply by Una865** [reply]
!python detect.py --weights runs/train/exp2/weights/best.pt --img 416 --conf 0.1 --max_det 1 --source '/content/American-Sign-Language-Letters-1/test/images'
This does not quite works

**Reply by glenn-jocher** [reply]
@Una865 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to help us better understand and diagnose your problem. 

Thank you! 😃
